Implement projects as groups

Arshdeep54 opened the pull request.

Arshdeep54 committed:
Add custom logs

Arshdeep54 committed:
correct log file path

Arshdeep54 committed:
fix cargo test

Arshdeep54 committed:
fix log implementation

Arshdeep54 committed:
adds debug statements as logs

Arshdeep54 committed:
add debug flag in logging

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Implement projects as groups

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add watchdog update function

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Restrict user to only login to his account

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update pam_ruser to pam_user

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update validate user function

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

DarkPhoenix42 commented on src/lib/utils.rs:
```diff
@@ -1,16 +1,56 @@
     Ok(())
 }
 
+pub fn add_user_to_groups(user: &str, groups: &[String]) -> Result<()> {
```
Can this function be changed so that it is OS agnostic?

Arshdeep54 replied:
Moved to watchdog_utils

DarkPhoenix42 commented on src/lib/logger.rs:
```diff
@@ -0,0 +1,61 @@
+use crate::config::read_config;
+use crate::utils::parse_offset;
+
+pub fn log(filepath: &str, status: &str, message: &str) -> Result<()> {
```
I think using a log library would be better instead of writing our own logger. That way we can also have fine control over the required log level.

DarkPhoenix42 commented on src/update.rs:
```diff
@@ -0,0 +1,44 @@
+
+pub fn handle_update() -> Result<()> {
+    let mut users: Vec<String>=  Vec::new();
+    logger::logln("in handle_update");
```
We also wouldn't need these lines if we use a logging crate.

DarkPhoenix42 commented on src/lib/utils.rs:
```diff
@@ -1,16 +1,56 @@
+use crate::{errors::*, logger};
 
-use crate::errors::*;
+pub const AUTH_LOG_PATH: &str = "/opt/watchdog/custom-logs/auth.logs";
```
I think these should be defined in another file, as they are not utilities.

DarkPhoenix42 commented on src/ssh.rs:
```diff
@@ -4,40 +4,46 @@ use std::process::Command;
-use lib::keyhouse::get_name;
 use lib::notifier;
-use lib::utils::clear_file;
+//use lib::utils::clear_file;
```
nit: remove the unused `use` statement.

DarkPhoenix42 commented on Cargo.toml:
```diff
@@ -16,6 +16,9 @@ clap = "2.19"
 toml_edit="0.1.5"
 nix = "0.9.0"
+openssl = "0.10"
+openssl-sys = "0.9.58"
```
I do not see these used anywhere. Is there a reason for including openssl?

Arshdeep54 replied:
Some module/dependency needed it, the build was failing, 
This is removed in latest commit

DarkPhoenix42 commented on src/lib/keyhouse.rs:
```diff
@@ -90,3 +136,97 @@ pub fn get_name(config: &Config, ssh_key: &str) -> Result<String> {
+                logger::logln(&format!("Projects: {:?}", projects));
+                Ok(projects)
+            } else {
+                logger::logln(&format!(
```
This logging should ideally be done by the caller, after handling the returned error. Both logging it and returning the same error does not make sense.

DarkPhoenix42 commented:
nit: an empty line separating relevant sections of code would be nice to enhance readability. Also, it seems to me that some files have not been formatted. Although, that might just be because we use different formatting settings.

Arshdeep54 committed:
Update ssh message with threads

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Implement logging with fern

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add logs filtering

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update handle_update to add new users

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update logs and create_user function

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Fix fetching from master branch

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add constant.rs and enums

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update user's bashrc with group's bashrc

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update add to sudo and wheel group

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add auto-update feature in handle_auth

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Fix multiple appends in .bashrc

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Log sudo command on slack

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Update based on last commit

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Fix base_url in handle_update

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Fix verbose logging

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add verbose level in config

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add default logs to watchdog.logs

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Add dependencies logs in dependencies.logs

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Fix HOME env in update_user_bashrc

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

riteshco committed:
Fix(sudo): sudo command logs on slack

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

Fix: extract_sudo_command

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

Fix: code structure

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

add three different methods to find the sudo command

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

refined three methods approach for sudo command extraction

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

fix: extract sudo two method approach

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

cargo error fixes

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

edited method 2

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

testing function sudo_extract_command

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

add testing command to check

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

cargo fmt run

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

add logging to identify the successfull sudo extract command method

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

add debugging logs in sudo extract command in utils file

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

cargo fmt and clippy on whole codebase

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

reverting last changes

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

pr changes

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>

satindra-r committed:
add build files for AUR

satindra-r committed:
add build files for Debian

Arshdeep54 committed:
Merge pull request #29 from sdslabs/sudo-regie

Fix sudo command extraction

satindra-r committed:
feat: Move repeated logic in scout to watchdog-utils

Arshdeep54 committed:
Merge pull request #30 from satindra-r/saturn/watchdog-utils

Use watchdog utils

Arshdeep54 committed:
Merge pull request #32 from satindra-r/saturn/AUR

add build files for AUR

Arshdeep54 committed:
chore: remove unused functions

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
chore: fix and update CI

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

riteshco committed:
add local cache while updating and authentication (#35)

add caching check during auth process and use watchdog utils' caching
logic during update
cache path in config variables
stable for testing (providing cache_path to watchdog utils)
to test on centos
undo testing changes
move check_local_cache from auth.rs to lib/utils.rs
config.rs changes for PathBuf
directly sending keyhouse struct to watchdog-utils in handle_update
remove warn for error logs

Signed-off-by: riteshco <riteshgadwalv2@gmail.com>
Co-authored-by: Satindra Ramalingam <satindra.r@gmail.com>
Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

satindra-r committed:
add --all flag for watchdog update

Arshdeep54 committed:
refactor(ci): run CI for all branches

Signed-off-by: Arshdeep54 <balarsh535@gmail.com>

Arshdeep54 committed:
Merge pull request #36 from satindra-r/saturn/--all

add --all flag

satindra-r committed:
bump watchdog_utils version

Arshdeep54 committed:
Merge pull request #39 from satindra-r/saturn/user-dir

satindra-r committed:
switch to git (#40)

satindra-r committed:
send all four retrived commands to slack (#42)
